airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -709,9 +711,9 @@ def create_simple_retriever(self, model: SimpleRetrieverModel, config: Config, *
                 parameters=model.parameters,
             )
         return SimpleRetriever(
-            name=model.name,
+            name=name,
             paginator=paginator,
-            primary_key=model.primary_key.__root__ if model.primary_key else None,
```

Comment:
This was unrelated to your exact change, but during the review I noticed that we're not actually using the `PrimaryKey` definition in our schema for `DeclarativeStream.primary_key`. See here: https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml#L890-L902

For consistency, can we make it point to the `PrimaryKey` definition. And then we'll need to keep this ternary operator to properly resolve the value. Although maybe we can do that higher up in create_declarative_stream()` before we pass it to the children. wdyt?

Reply:
Makes sense to me! Updated.